pkg/client/observatorium/observability_config.go

```diff
@@ -80,6 +70,74 @@ type ObservabilityCloudWatchLoggingConfig struct {
 	configFilePath                string                                                      `validate:"-"`
 }
 
+type DataPlaneObservabilityOIDCCredentials struct {
+	ClientID     string `yaml:"client_id" validate:"required"`
+	ClientSecret string `yaml:"client_secret" validate:"required"`
+}
+
+type DataPlaneObservabilityOIDCConfiguration struct {
+	AuthorizationServer string                                 `yaml:"authorization_server" validate:"required"`
+	Realm               string                                 `yaml:"realm" validate:"required"`
+	Credentials         *DataPlaneObservabilityOIDCCredentials `yaml:"credentials" validate:"required,dive"`
+}
+
+type DataPlaneObservabilityRemoteWriteConfiguration struct {
+	RemoteWriteUrl    string                                   `yaml:"remote_write_url" validate:"required"`
+	OIDCConfiguration *DataPlaneObservabilityOIDCConfiguration `yaml:"oidc_configuration" validate:"omitempty,dive"`
+}
+
+type DataPlaneObservabilityConfig struct {
+	configFilePath string
+	Enabled        bool
+
+	RemoteWriteConfiguration *DataPlaneObservabilityRemoteWriteConfiguration `yaml:"remote_write_configuration" validate:"required,dive"`
+	GithubResourcesAuthToken string
+}
+
+func (c *DataPlaneObservabilityConfig) GetOIDCClientID() string {
```

Comment:
I see lots of checks where `c.Enabled` is checked spread around several methods.
I think this might be overly complicated. Couldn't we only use those methods if it is enabled? That is, DataPlaneObservabilityConfig is only used by the clients if it is enabled

Reply:
we could, but I would prefer to keep the use of those methods simple with all the checks being opaque to the caller. For example, we use them to assign fields in a secret in cluster_mgr.go. We would have to move the enabled check there.

Reply:
With what I propose we make the `DataPlaneObservabilityConfig` methods simpler instead of spreading them among all the methods of it.
Then on the client side the only thing you do is to check if it is enabled and if it is then you use it.
DataPlaneObservabilityConfig methods have the precondition that it will be enabled (other than the read and validate one)

Reply:
@pb82 @miguelsorianod there is this check https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/3cb3aa93e8989c6654abc741ab883772fcf0c0b4/internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go#L783 that I already see, my thinking is that is enough to control the creation of the service account secret. And we do not need the extra `c.Enabled` checks in the methods in the config

Reply:
> @pb82 @miguelsorianod there is this check
> 
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/3cb3aa93e8989c6654abc741ab883772fcf0c0b4/internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go#L783
> that I already see, my thinking is that is enough to control the creation of the service account secret. And we do not need the extra `c.Enabled` checks in the methods in the config

+1. Or combine with Enabled && HasOIDCConfiguration better

Reply:
Yes @pb82 but only if you do not want the secret key to be present.
The current behaviour is that an `""` secret is going to be created which I think can still be achieved by removing the enabled check. To be clear, the nillness check has to stay. In that case, we'll continue to send `""` string.

Reply:
i.e send an `""` string if the underlying value is empty in the configuration or the configuration is `nil`. Which is what we intend to do.

Reply:
Ok, let me refactor this then.